Bring into compliance with WILDS rules

sckott opened the pull request:
## Rules and changes here in this PR

- At least two but no more than three designated project leads (specified in the [CODEOWNERS file](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners)).
  - commit f63933cac78c3a2f61fbb52a1db932480a99ee91
  - note that "Code owners are automatically requested for review when someone opens a pull request that modifies code that they own." - presumably this is okay? do you want to be listed or no? that is do you want to be automatically asked for PR review? 
- In all repositories in WILDS we will include mention of our employer  https://getwilds.org/guide/conventions.html#mention-our-employer 
  - commits f929faeac9866e443365c2ac5d3e70e020f7f869 and e3cb2160f95f5b5d42179882a10dba646f1bed88
 
## Rules we're not currently following

- `main` branch protections such that `main` can only be updated by a pull request from `dev` or a branch that begins with `hotfix-`, and a review from at least one project lead.

Should we? We say that every project with Prototype status and above should have this

## Other rules to follow with the upcoming release

- Every merge from `dev` into `main` should constitute a release, which should generate a tagged version of the software and an increment to the version number

seankross commented:
> presumably this is okay? do you want to be listed or no? that is do you want to be automatically asked for PR review?

That is okay, I do want to be automatically asked for PR review.

> Should we? We say that every project with Prototype status and above should have this

We should.

sckott commented on DESCRIPTION:
```diff
@@ -3,9 +3,10 @@ Title: Humane Interface to AWS
+    person("Scott", "Chamberlain", , "sachamber@fredhutch.org", role = c("aut", "cre"),
+           comment = c(ORCID = "0000-0003-1444-9135")),
+    person("Fred Hutchinson Cancer Center", , , "wilds@fredhutch.org", role = "fnd",
+           comment = c(ROR = "007ps6h72"))
```
yeah it's very recent. so recent in fact to submit to cran right now you have to build the package to submit with R-devel 

example here with vcr https://cloud.r-project.org/web/packages/vcr/ - it adds a little image and links to the ROR page for it

sckott commented:
@seankross 

- codeowners: okay, great. i'll leave you in 
- branch protections: okay, so do you want to do that:
  - at the getwilds org level (in the org settings)
  - in the settings for this repo
  - maybe we can do in the codeowners file itself, not sure though

seankross commented:
> branch protections

sckott commented:
okay i'll add the branch protection to this repo only

sckott commented:
@seankross branch protection info, please review:

- added new GH action workflow `.github/workflows/protect-main.yaml` - it checks the branch name for the PR and fails if it's not `dev` or `hotfix-*`, and passes if it is one of those. The purpose of this workflow is just to check branch names, whereas the next bullet is about enforcing other stuff. See e.g., failure https://github.com/getwilds/sixtyfour/actions/runs/13816227962/job/38650023133?pr=115 
- also added a branch ruleset in the Settings tab here (take a look yourself). It 
  - applies to the default branch (main in our case)
  - bypass permissions for: org admins, repo admins
  - only allows users with bypass permissions to delete branches
  - requires a PR before merging with at least 1 approval
  - require this new protect main status check to pass (we could require all checks or at least more than this one, but that seems maybe not great since sometimes there's check failures that can't be addressed until merging) - This isn't being enforced yet, possibly because we're not merging into main, not sure
  - blocks force pushes

seankross commented on .github/workflows/protect-main.yaml:
```diff
@@ -0,0 +1,28 @@
+  pull_request:
+    branches:
+      - main
+      - dev
```
Should this only be main or main and dev?

sckott replied:
sorry, the `dev` here was just so that the action would run in this PR - i'll remove after we're happy with it

any other thoughts?

seankross reviewed (approved):
Thank you!
